src/popup/index.ts

```diff
@@ -54,19 +181,30 @@ browser.storage.onChanged.addListener((changes) => {
   }
 });
 
-function handleStartTimerButton() {
-  playSound("button");
-  handleStartTimer();
-}
-
-function handleTimerSelect(e: Event) {
-  const selectElement = e.target as HTMLSelectElement;
-  const minutes = parseInt(selectElement.value);
-  changeSelectedTime(minutes);
-
-  timer.innerHTML = `${minutes <= 9 ? "0" + minutes : minutes}:00`;
-}
-
 startButton.addEventListener("click", handleStartTimerButton);
 selectTime.addEventListener("change", handleTimerSelect);
 configButton.addEventListener("click", () => browser.runtime.openOptionsPage());
+
+timerDisplay.addEventListener("click", () => {
+  if (!isTimerRunning) {
+    customInput.style.display = "block";
+    timerDisplay.style.display = "none";
+    customInput.value = timerDisplay.textContent || "00:00";
+    customInput.focus();
+  }
+});
+
+customInput.addEventListener("blur", applyCustomTime);
+customInput.addEventListener("keypress", (event) => {
+  if (event.key === "Enter") {
+    applyCustomTime();
+  }
+});
+
+startButton.addEventListener("click", () => {
+  isTimerRunning = !isTimerRunning;
+  startButton.textContent = isTimerRunning ? "STOP" : "START FOCUSING";
```

Comment:
Hey @gabireze, when the start button is pressed the text will change to "STOP" instead of "GIVE UP!",  only after the counter delay will the text change to  "GIVE UP!".

Changing "STOP" to "GIVE UP!" fixes it.

Reply:
I've made the correction. Thank you for informing me!